packages/apollo-language-server/src/utilities/graphql.ts

```diff
@@ -105,18 +105,19 @@ export function removeDirectives<AST extends ASTNode>(
 }
 
 /**
- * Recursively remove orphaned fragment definitions that have their names included in
- * `fragmentNamesEligibleForRemoval`
+ * Recursively remove orphaned fragment definitions and variables that have their names included in
+ * `fragmentNamesEligibleForRemoval` and `variablesEligibleForRemoval`, respectively.
  *
  * We expclitily require the fragments to be listed in `fragmentNamesEligibleForRemoval` so we only strip
  * fragments that were orphaned by an operation, not fragments that started as oprhans
  *
  * The `ast` param must extend ASTNode. We use a generic to indicate that this function returns the same type
  * of it's first parameter.
  */
-function removeOrphanedFragmentDefinitions<AST extends ASTNode>(
+function removeOrphanedFragmentDefinitionsAndVariables<AST extends ASTNode>(
```

Comment:
It might be reasonable to split this function into two. I was trying to avoid running through the AST too many times.